File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.java
class Solution {
    public int solution(int[] num_list) {
        StringBuilder odd = new StringBuilder();
        StringBuilder even = new StringBuilder();
        
        for (int i = 0; i < num_list.length; i++) {
            if (num_list[i] % 2 == 0) {
                even.append((char)(num_list[i] + '0'));
            } else {
                odd.append((char)(num_list[i] + '0'));
            }
        }
        
        return Integer.parseInt(odd.toString()) + Integer.parseInt(even.toString());
    }
}

File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> num_list) {
    string odd = "";
    string even = "";
    
    for (int num : num_list) {
        if (num % 2 == 0) even.push_back(num + '0');
        else odd.push_back(num + '0');
    }
    
    return stoi(odd) + stoi(even);
}